crates/containerd-shim-wasm/src/sandbox/instance_utils.rs

```diff
@@ -45,6 +46,38 @@ pub fn maybe_open_stdio(path: &str) -> Result<Option<File>, Error> {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+struct Options {
+    root: Option<PathBuf>,
+}
+
+pub fn determine_rootdir<P: AsRef<Path>>(
+    bundle: P,
+    namespace: &str,
+    container_root_dir: P,
+) -> Result<PathBuf, Error> {
+    log::info!(
+        "determining rootdir for bundle: {}",
+        bundle.as_ref().display()
+    );
+    let mut file = match File::open(bundle.as_ref().join("options.json")) {
+        Ok(f) => f,
+        Err(err) => match err.kind() {
+            ErrorKind::NotFound => return Ok(container_root_dir.as_ref().join(namespace)),
+            _ => return Err(err.into()),
+        },
+    };
+    let mut data = String::new();
+    file.read_to_string(&mut data)?;
+    let options: Options = serde_json::from_str(&data)?;
+    let path = options
+        .root
+        .unwrap_or(container_root_dir.as_ref().to_path_buf())
+        .join(namespace);
+    log::info!("youki root path is: {}", path.display());
```

Comment:
since this is specific to youki, could it be moved into the libcontainer_instance folder?

Reply:
Good point

Reply:
Actually this might not be specific to youki...

Reply:
Essentially, it uses options.json to find the root path of the container runtime (e.g. in wasmtime case, the default is `/run/containerd/wasmtime`).

Reply:
Yes, I think this is not youki's spec.